### DISCUSSION TITLE
how to create (and populate) a project from the command line

### DISCUSSION THREAD

**Discussion by AlessandroLas** [broadinstitute/seqr#2998] [Q&A] [answered]
Hi everyone.
In my institute we are testing a local install of seqr.
I'm new to the tool and I like to ask if there is the possibility to 
get a project created "in its main parts" from the command line.
I mean, I can move a vcf to the input_vcf directory and launch the annotation (and see the index in the elasticsearch).
I can move the matched bam file in the readviz directory. I can create the necessary files for creating the project (families, individuals, path_to_bam, ...).
However, I couldn't find instructions on how to create (and populate) a project from the command line.
Subsequently, I'd also like to learn how to assign groups and permissions for that new project.

**Comment by hanars** [comment, marked as answer]
There is no command line way to populate a project. You need to log into the application and use the "create project" button at the bottom of the main dashboard page, and then once you do that go to the project page and use the "add collaborator" button to add users to the project, and the "add families and individuals" button to add pedigree information. Once your pipeline run completes and you have added your pedigree to the project, you will use the "add datasets" button on the project page to update the project to reference the new index

**Comment by ghost** [comment]
Our team actually has a bunch of tools that we created to automate our project creation from the command line if this is still of interest and not redundant:
```bash
seqr="manage.py"
# Create project
$seqr create_project -project_name BATCH9001 -genome_version 38 -categories private WGS
# Adding families to the project from a PED file
$seqr ped_bulk_upload -ped BATCH9001.ped -project BATCH9001 -user gilmorera
# Loading the CRAM file paths for IGV
$seqr upload_bam_paths -f BATCH9001.CramPath_forseqr.tsv -project BATCH9001
# Associating the index with the project
$seqr upload_callset -project BATCH9001 -index best_index -type VARIANTS -ignore yes
# Adding our tags
$seqr add_project_tag_from_file BATCH9001
# Adding our team for batch QC/prerelease
$seqr manage_users_in_project_from_file -action add -project_name BATCH9001 -file GRIS_team.txt
# Adding our gene lists
$seqr add_genelists_in_project_from_file -project BATCH9001 -file Default_Gene_Lists.txt
# updating the redis cache
$seqr update_individuals_cache -p BATCH9001
$seqr update_families_cache -p BATCH9001
```

There are a bunch more, but I thought I would stay on topic.

**Reply by hanars** [reply]
While we have no issues with your group taking this approach, this is not the approach we are generally recommending for seqr use so I do not think our group would want to include these commands as a core part of seqr.  The intended goal for seqr is that it is initially set up by engineers, but that all day-to-day operations can be handles by non-technical users (although we are well aware that we aren't quite there yet)

**Comment by AlessandroLas** [comment]
Dear Robert,
Thank you for your reply.

Actually I'm still interested in finding ways to automate those tasks.
If I understand correctly, you added pieces of code to manage.py.
I would like to thank you if you want to share your tools.
Do you have a github for that?

Alessandro.


Il giorno ven 16 feb 2024 alle ore 16:08 Robert Gilmore <
***@***.***> ha scritto:

**Comment by ghost** [comment]
Hi @AlessandroLas.  We don't currently have a public github for these tools, but I will relay to my team that you and others might be interested in this type of functionality and follow up with you here.

**Comment by AlessandroLas** [comment]
Many thanks in advance, Robert.


Il giorno lun 19 feb 2024 alle ore 14:53 Robert Gilmore <
***@***.***> ha scritto:
